web/modules/custom/server_general/src/Plugin/EntityViewBuilder/NodeGroup.php

```diff
@@ -0,0 +1,212 @@
+class NodeGroup extends NodeViewBuilderAbstract {
+
+  use RedirectDestinationTrait;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The OG access service.
+   *
+   * @var \Drupal\og\OgAccessInterface
+   */
+  protected $ogAccess;
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new nodegroup object.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user.
+   * @param \Drupal\og\OgAccessInterface $og_access
+   *   The OG access service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   */
+  public function __construct(AccountInterface $current_user, OgAccessInterface $og_access, EntityTypeManagerInterface $entity_type_manager) {
+    $this->currentUser = $current_user;
+    $this->ogAccess = $og_access;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+          $container->get('current_user'),
+          $container->get('og.access'),
+          $container->get('entity_type.manager')
+      );
+
+  }
+
+  /**
+   * Build full view mode.
+   *
+   * @param array $build
+   *   The existing build.
+   * @param \Drupal\node\NodeInterface $entity
+   *   The entity.
+   *
+   * @return array
+   *   Render array.
+   */
+  public function buildFull(array $build, NodeInterface $entity): array {
+
+    // Cache by the OG membership state. Anonymous users are handled below.
+    $build['#cache']['contexts'] = [
+      'og_membership_state',
+      'user.roles:authenticated',
+    ];
+    $cache_meta = CacheableMetadata::createFromRenderArray($build);
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $cache_meta->merge(CacheableMetadata::createFromObject($entity));
+    $cache_meta->applyTo($build);
+
+    $user = $this->entityTypeManager->getStorage('user')->load(($this->currentUser->id()));
+
+    if ($entity->getOwnerId() == $user->id()) {
+      // User is the group manager.
+      $build[] = [
+        '#type' => 'html_tag',
+        '#tag' => 'span',
+        '#attributes' => [
+          'title' => $this->t('You are the group manager'),
+          'class' => ['group', 'manager'],
+        ],
+        '#value' => $this->t('You are the group manager'),
+      ];
+
+      return $build;
+    }
+
+    $storage = $this->entityTypeManager->getStorage('og_membership');
+    $props = [
+      'uid' => $user ? $user->id() : 0,
+      'entity_type' => $entity->getEntityTypeId(),
+      'entity_bundle' => $entity->bundle(),
+      'entity_id' => $entity->id(),
+    ];
+
+    $memberships = $storage->loadByProperties($props);
```

Comment:
Let's use OG's API to get that

Reply:
Cloud you add more information here

Reply:
Use OG's API to get the membership

Reply:
used OG's API to get the membership. please check